### PR TITLE
JS-20: Add more complex null value handling for v2

### PR DESCRIPTION
**Pull request by mvrueden**
This adds more complex null value handling to the V2FilterProcessor:
* JIRA: https://issues.opennms.org/browse/JS-20

This will still not fully work, due to the following issue: 
* https://issues.opennms.org/browse/NMS-9578

### CODE REVIEW

File: src/dao/AbstractDAO.ts [hunk 1]
```diff
@@ -154,19 +178,24 @@ export abstract class AbstractDAO<K, T> {
...
+          .then((options) => {
+              if (this.useJson()) {
+                  options.headers.accept = 'application/json';
+              } else {
+                  // always use application/xml in DAO calls when we're not sure how
+                  // usable JSON output will be.
+                  options.headers.accept = 'application/xml';
+              }
+              if (filter) {
+                  return this.getFilterProcessor().then((processor) => {
```
Comment (RangerRick) [inline src/dao/AbstractDAO.ts]: You could avoid having to refactor all of the `getOptions` code into async by using `await` here and making it synchronous again.

**Comment by RangerRick** [comment]
Also it would be nice if the code to create the properties cache was a separate commit from changing the date parsing stuff, having it all together makes it kind of hard to tell what's going on.

File: src/dao/AbstractDAO.ts (hunk 1 above)
Comment (mvrueden) [inline src/dao/AbstractDAO.ts]: You can only use await in an async-function.

File: src/internal/Util.ts [hunk 2]
```diff
@@ -69,4 +69,15 @@ export class Util {
...
     }
   }
+
+  /**
+   * Encodes the parameters.
+   * @param pramaters the parameters to encode
+   */
+  public static encodeParameters(pramaters: any) {
+      Object.keys(pramaters).forEach((key, index) => {
+          pramaters[key] = encodeURIComponent(pramaters[key]);
```
Comment (RangerRick) [inline src/internal/Util.ts]: `encodeURIComponent` is the right way to do this, yeah.  I did some digging for the Axios side and it does appear that they don't do it for you, so it should be correct to encode it for all implementations.

File: src/dao/AbstractDAO.ts (hunk 1 above)
Comment (RangerRick) [inline src/dao/AbstractDAO.ts]: Ah, dang, that's right.  Never mind then.  ;)

**Comment by mvrueden** [comment]
Yeah, you are right. I'll keep that in mind for the next time (-:

**Pull request merged**